### DISCUSSION TITLE
Some help

### DISCUSSION THREAD

**Discussion by MarkKlerkx** [nrubin29/finale#28] [General]
Hello Noah,

Maybe this is not the place to ask you this question.

I am not a developper but I like to play around with code, so I am busy to develop a website with javascript for browsers to recognize music and give me a lot of infornation. The purpose is to have the Roon/spotify experience when playing music on vinyl.

When I use my windows laptop almost all songs were recognized. When I use my Android mobile devices none were recognized. I noticed that the recording quality of the sample was bad.

But your app almost recognizing everything on the same Android devices.

My question is; did you add some code to upgrade the quality of the sample?

Thank you in advance!

Mark

**Comment by nrubin29** [comment]
Hi Mark,

Finale just uses the [ACRCloud](https://www.acrcloud.com) framework via a Flutter plugin I wrote called [flutter_acrcloud](https://github.com/nrubin29/flutter_acrcloud). I didn't do anything special to make the quality better. Are you sure the the Android device is actually getting audio? Note that you have to ask for permission to use the microphone and the simulator doesn't have a microphone attached.

Cheers,
Noah

**Comment by MarkKlerkx** [comment]
Thanks for your quick reply on my question!

The website is asking for permissions both on the Windows device and on Android/iOS.
I made a test option to record audio instead of sending it to ACRCloud, and I noticed that the quality of the sample from an Android device sucks. The microphone is clipping, a lot of roomspace noise. So I can understand that ACRCloud cannot recognize any music out of it 😊

But when I do a record with the Android recorder or Whatsapp it sounds a lot better.
So there seems to be a difference in recording audio from a website (by Javascript) or from a native Android app.

With kind regards,

Mark

**Comment by nrubin29** [comment]
Here's some potentially relevant Android code from my `flutter_acrcloud` plugin: https://github.com/nrubin29/flutter_acrcloud/blob/master/android/src/main/kotlin/com/nrubintech/flutter_acrcloud/FlutterAcrcloudPlugin.kt#L100

I wonder if changing some values in the `recorderConfig` would result in a better recording. I personally don't use an Android device so I can't really test it, but if you're able to and you can get it to produce a recording with better quality, please let me know.
